src/packages/@environments/features/environment-list/layout/EnvironmentList.svelte

```diff
@@ -0,0 +1,255 @@
+      </p>
+    {/if}
+    <hr class="mb-0 mt-1" />
+
+    <p class="fw-normal text-fs-14 rounded my-2" style="padding-left: 12px;">
+      Environment Variables
+    </p>
+    {#if localEnvironment && localEnvironment.length === 0}
+      <div class={`pt-3 pb-2`}>
+        <p class={`add-env-desc-text text-fs-12 mb-0 fw-normal text-center`}>
+          Add Environments to your Workspace to test your APIs with the relevant
+          set of resources and constraints.
+        </p>
+      </div>
+    {/if}
+    <ul class={`env-side-tab-list p-0`}>
+      <List height={"calc(100vh - 180px)"} classProps={"pb-2 pe-2"}>
+        {#if localEnvironment && localEnvironment.length > 0}
+          {#each localEnvironment as env}
+            <ListItem
+              {env}
+              {currentWorkspace}
+              {currentEnvironment}
+              {onDeleteEnvironment}
+              {onUpdateEnvironment}
+              {onOpenEnvironment}
+              {onSelectEnvironment}
+            />
+          {/each}
+        {/if}
+        <p
+          class="fw-normal d-flex align-items-center text-fs-14 m-2 ms-3"
+          role="button"
+          on:click={() => {
+            onCreateEnvironment(localEnvironment);
+          }}
+        >
+          <PlusIcon />
+          <span>Add Environment</span>
+        </p>
+      </List>
+    </ul>
+  {/if}
+</div>
+
+<style lang="scss">
+  .navbar {
+    height: auto;
+    overflow: hidden;
+  }
+  ul li {
+    display: block;
+  }
+
+  ul li button {
+    font-size: 12px;
+    display: flex;
+    width: 100%;
+    border: 0px;
+    background-color: var(--blackColor);
+  }
+
+  ul li button:hover {
+    width: 100%;
+    color: var(--white-color);
+    border-radius: 8px;
+    background-color: #232527;
+  }
+  .env-sidebar {
+    height: calc(100vh - 44px);
+  }
+  .curr-workspace {
+    max-height: 20vw;
+  }
+  .add-env-mini-btn {
+    background-color: transparent;
+  }
+  .add-env-mini-btn:hover {
+    background-color: var(--border-color);
+  }
+  .add-env-mini-btn:active {
+    background: var(--workspace-hover-color);
+  }
+  .add-env-desc-text {
+    color: #999;
+  }
+  .add-env-btn {
+    background: linear-gradient(270deg, #6147ff -1.72%, #1193f0 100%);
+  }
+  .env-item {
+    padding: 6px 6px 6px 12px;
+  }
+  .env-item:hover {
+    background-color: var(--bg-secondary-800);
+  }
+  .env-item.active {
+    background-color: var(--bg-tertiary-600);
+  }
+  .env-side-tab-list {
+    list-style: none;
+    overflow: none;
+  }
+
+  .show-more-btn {
+    margin-right: 0;
+    background-color: transparent;
+    visibility: hidden;
+  }
+
+  .env-item:hover .show-more-btn {
+    visibility: visible;
+  }
+  .show-more-btn.active {
+    visibility: visible;
+  }
+  .rename-input {
+    background-color: transparent;
+  }
+  .rename-input:focus {
+    outline: none;
+    background-color: var(--border-color);
+    border-bottom: 1px solid #85c2ff !important;
+  }
+  .spinner {
```

Comment:
is this right place for spinner style? we are already having a component for `Spinner` that we are using. Are these additional parameters that we need to configure for it? if yes, then these should be spinner props. And if we talk about positioning, we can still use bootstrap classes rather than writing custom css.

Reply:
resolved